hummingbot/connector/exchange/bybit/bybit_exchange.py

```diff
@@ -0,0 +1,887 @@
+                min_price_increment = rule.get("minPricePrecision")
+                min_base_amount_increment = rule.get("basePrecision")
+                min_notional_size = rule.get("minTradeAmount")
+
+                retval.append(
+                    TradingRule(trading_pair,
+                                min_order_size=Decimal(min_order_size),
+                                min_price_increment=Decimal(min_price_increment),
+                                min_base_amount_increment=Decimal(min_base_amount_increment),
+                                min_notional_size=Decimal(min_notional_size)))
+
+            except Exception:
+                self.logger().exception(f"Error parsing the trading pair rule {rule.get('name')}. Skipping.")
+        return retval
+
+    async def _user_stream_event_listener(self):
+        """
+        This functions runs in background continuously processing the events received from the exchange by the user
+        stream data source. It keeps reading events from the queue until the task is interrupted.
+        The events received are balance updates, order updates and trade events.
+        """
+        async for event_message in self._iter_user_event_queue():
+            try:
+                event_type = event_message.get("e")
+                if event_type == "executionReport":
+                    execution_type = event_message.get("X")
+                    client_order_id = event_message.get("c")
+                    if execution_type in ["PARTIALLY_FILLED", "FILLED"]:
+                        tracked_order = self._order_tracker.fetch_order(client_order_id=client_order_id)
+                        if tracked_order is not None:
+                            fee = TradeFeeBase.new_spot_fee(
+                                fee_schema=self.trade_fee_schema(),
+                                trade_type=tracked_order.trade_type,
+                                flat_fees=[TokenAmount(amount=Decimal(event_message["n"]), token=event_message["N"])]
+                            )
+                            trade_update = TradeUpdate(
+                                trade_id=str(event_message["E"]),
+                                client_order_id=client_order_id,
+                                exchange_order_id=str(event_message["i"]),
+                                trading_pair=tracked_order.trading_pair,
+                                fee=fee,
+                                fill_base_amount=Decimal(event_message["l"]),
+                                fill_quote_amount=Decimal(event_message["l"]) * Decimal(event_message["L"]),
+                                fill_price=Decimal(event_message["L"]),
+                                fill_timestamp=int(event_message["E"]) * 1e-3,
+                            )
+                            self._order_tracker.process_trade_update(trade_update)
+
+                    tracked_order = self.in_flight_orders.get(client_order_id)
```

Comment:
There is already logic to get the tracked order in line 722. That line should be moved up to make the info available for this logic too, instead of looking two times for the same information.